compiler/command_categories.py

```diff
@@ -104,6 +106,31 @@ def __repr__(self):
         return "Aggregator(name={},opts={})".format(self.name, self.options)
 
 
+## The class that contains the mapper information that was mapped from annotation (rare).
+class Mapper:
+    def __init__(self, mapper_json):
+        ## Exactly one of the two should exist in the JSON
+        assert('name' in mapper_json or 'rel_path' in mapper_json)
+        assert(not ('name' in mapper_json and 'rel_path' in mapper_json))
+        
+        if('rel_path' in mapper_json):
+            ## Set the name to be the absolute path
+            self.name = os.path.join(config.PASH_TOP, mapper_json['rel_path'])
```

Comment:
This is great — it looks like someone can use an absolute path (for testing, if needed), and the resolution would still work correctly! e.g., `os.path.join("/home/nikos", "/Users/nikos/pizza")` would return `'/Users/nikos/pizza'`. So maybe instead of `rel_path`, we could just name it `path`.

Reply:
@angelhof